applications/ParticleMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -469,28 +469,9 @@ void UpdatedLagrangian::CalculateAndAddRHS(
         : false;
     if (is_explicit)
     {
-        Matrix Jacobian;
-        GetGeometry().Jacobian(Jacobian, 0);
-        Matrix InvJ;
-        double detJ;
-        MathUtils<double>::InvertMatrix(Jacobian, InvJ, detJ);
-        const Matrix& r_DN_De = GetGeometry().ShapeFunctionLocalGradient(0);
-        rVariables.DN_DX = prod(r_DN_De, InvJ); // cartesian gradients
-
-        const bool is_axisymmetric = (rCurrentProcessInfo.Has(IS_AXISYMMETRIC))
-            ? rCurrentProcessInfo.GetValue(IS_AXISYMMETRIC)
-            : false;
-
-        if (is_axisymmetric) {
-            const double current_radius = ParticleMechanicsMathUtilities<double>::CalculateRadius(
-                GetGeometry().ShapeFunctionsValues(), GetGeometry());
-            MPMExplicitUtilities::CalculateAndAddAxisymmetricExplicitInternalForce(*this,
-                rVariables.DN_DX, mMP.cauchy_stress_vector, mMP.volume,
-                mConstitutiveLawVector->GetStrainSize(), current_radius, rRightHandSideVector);
-        }
-        else MPMExplicitUtilities::CalculateAndAddExplicitInternalForce(*this,
-            rVariables.DN_DX, mMP.cauchy_stress_vector, mMP.volume,
-            mConstitutiveLawVector->GetStrainSize(), rRightHandSideVector);
+        MPMExplicitUtilities::CalculateAndAddExplicitInternalForce(*this,
```

Comment:
this could be done externally now, right?

Reply:
Yep, I'm pretty sure it could all be done from the scheme